Arrowgene.Ddon.GameServer/Characters/JobManager.cs

```diff
@@ -193,6 +195,37 @@ private List<CDataItemUpdateResult> SwapEquipmentAndStorage(GameClient client, C
                 {
                     // Equip item to a slot, if said item is not already equipped. Search item in multiple storages
                     List<CDataItemUpdateResult>? moveResult = null;
+
+                    //Search ahead for jewelry and remove in advance.
+                    if (JEWELRY_SLOTS.Contains((EquipSlot)equipSlot))
+                    {
+                        foreach (EquipSlot jewelrySlot in JEWELRY_SLOTS)
+                        {
+                            if ((byte)jewelrySlot <= equipSlot) continue;
+                            Item? oldJewelryItem = oldEquipment[(int)jewelrySlot - 1];
+                            if (oldJewelryItem != null && newEquipmentTemplateItem.UId == oldJewelryItem?.UId)
+                            {
+                                ushort laterSlot = common.Equipment.GetStorageSlot(equipType, (byte)jewelrySlot);
+                                try
+                                {
+                                    _Server.ItemManager.MoveItem(_Server, client.Character, common.Equipment.Storage, laterSlot, 1, client.Character.Storage.GetStorage(StorageType.StorageBoxNormal), 0);
+                                }
+                                catch (ResponseErrorException ex)
+                                {
+                                    if (ex.ErrorCode == ErrorCode.ERROR_CODE_CHARACTER_ITEM_NOT_FOUND)
```

Comment:
I dont like this pattern of error resolution. Why is this happening that an exception is being thrown and then we squelch it?

Reply:
The player no longer having an item they used to have when they last changed jobs is an expected case, it's not an error

Reply:
Why is there an exception then?

Reply:
Because we're reusing ItemManager for this which throws an exception when the item you're searching for is not present. Usually we want this to become an user facing error but here we don't

Reply:
This is completely the wrong way to use exceptions. If it's not an exceptional issue. Don't throw exceptions and handle it properly instead.